src/cli.cpp

```diff
@@ -344,6 +378,44 @@ int CLI::BINK2002() {
     return 0;
 }
 
+int CLI::BINK1998Validate() {
+    char product_key[PK_LENGTH]{};
+
+    if (CLI::stripKey(this->options.keyToCheck.c_str(), product_key) != 25) {
```

Comment:
I suggest moving the length check into the stripKey function, return false if a garbage char is encountered or the key is too long, return true otherwise